doxia-integration-tools/pom.xml

```diff
@@ -86,9 +86,15 @@
       <scope>provided</scope>
     </dependency>
     <dependency>
-      <groupId>org.apache.maven</groupId>
-      <artifactId>maven-plugin-api</artifactId>
-      <version>${mavenVersion}</version>
+      <groupId>org.eclipse.aether</groupId>
+      <artifactId>aether-api</artifactId>
+      <version>${aetherVersion}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.eclipse.aether</groupId>
+      <artifactId>aether-impl</artifactId>
```

Comment:
I don't think we need impl with that scope. If at all only with "test" scope.

Reply:
This is what `dependency:analyze` has told me. Try for yourself.

Reply:
It is a transitive dependency of `maven-core` and `maven-aether-provider` (the latter also being a transitive dependency of maven-core). I don't know how the <https://maven.apache.org/plugins/maven-dependency-plugin/analyze-mojo.htm>l with https://maven.apache.org/shared/maven-dependency-analyzer/ concludes that this is necessary (couldn't bring it to log the referencing class) but IMHO it isn't. @michael-o Any idea how to make the output of m-dependency-p more verbose in this regard?

Reply:
Does this work for you: https://maven.apache.org/plugins/maven-dependency-plugin/analyze-mojo.html#verbose?

Reply:
Either a bug in the plugin or these truly cause it: https://github.com/apache/maven-doxia-sitetools/blob/dacaa552c1b8e89eed84db0f43b6b0a72be91d0c/doxia-integration-tools/src/main/java/org/apache/maven/doxia/tools/DefaultSiteTool.java#L87-L96

Let me move it to test scope...

Reply:
Please move the test dependencies to the "test" scope section in the pom and also add the transitive excludes for `maven-core` (otherwise they are still transitive dependencies).

Reply:
I see this and `maven-core` is in `provided` scope, why should we exclude all deps? Moreover, we don't exclude in other components. I don't fully understand the reasoning.
Tree:
```
[INFO] +- org.apache.maven:maven-core:jar:3.2.5:provided
[INFO] |  +- org.apache.maven:maven-settings:jar:3.2.5:provided
[INFO] |  +- org.apache.maven:maven-settings-builder:jar:3.2.5:provided
[INFO] |  +- org.apache.maven:maven-repository-metadata:jar:3.2.5:provided
[INFO] |  +- org.apache.maven:maven-plugin-api:jar:3.2.5:provided
[INFO] |  +- org.apache.maven:maven-model-builder:jar:3.2.5:provided
[INFO] |  +- org.eclipse.aether:aether-util:jar:1.0.0.v20140518:provided
[INFO] |  +- org.sonatype.sisu:sisu-guice:jar:no_aop:3.2.3:provided
[INFO] |  +- org.codehaus.plexus:plexus-classworlds:jar:2.5.2:compile
[INFO] |  +- org.codehaus.plexus:plexus-component-annotations:jar:1.5.5:compile
[INFO] |  \- org.sonatype.plexus:plexus-sec-dispatcher:jar:1.3:provided
[INFO] |     \- org.sonatype.plexus:plexus-cipher:jar:1.4:provided
```

Reply:
To make m-dependency-p work correctly and to reduce the classpath of this module itself (it won’t have an effect on referencing projects as you noticed correctly). I consider this best practice given that Maven < 4 lacks a proper API artifact.

Reply:
@kwin Can we merge this as-is as a compromise like all other components just use `provided` scope?

Reply:
Sure, already an improvement as is, but I don't really get the argument against it (except for most of the other components do it like this).